app/src/main/java/co/nimblehq/ui/common/dialog/SubmitSurveyResponsesSuccessDialog.kt

```diff
@@ -0,0 +1,31 @@
+package co.nimblehq.ui.common.dialog
+
+import android.app.Dialog
+import android.graphics.drawable.ColorDrawable
+import android.os.Bundle
+import android.view.LayoutInflater
+import androidx.fragment.app.DialogFragment
+import co.nimblehq.R
+import co.nimblehq.data.lib.common.DEFAULT_DURATION
+import co.nimblehq.lib.EmptyCallback
+import java.util.*
+import kotlin.concurrent.schedule
+
+class SubmitSurveyResponsesSuccessDialog(private val callback: EmptyCallback? = null) : DialogFragment() {
+
+    override fun onCreateDialog(savedInstanceState: Bundle?): Dialog {
+        val dialog = Dialog(context, R.style.CenterFadeInOutDialogStyle)
+        dialog.window?.setBackgroundDrawable(ColorDrawable(android.graphics.Color.TRANSPARENT))
+        val view = LayoutInflater.from(context).inflate(R.layout.dialog_submit_survey_responses_success, null)
+        dialog.setContentView(view)
+        return dialog
```

Comment:
to implement a `DialogFragment`, I'd suggest you go through the same lifecycle as a normal fragment.
but if you would like to use this onCreateDialog callback I'd suggest rearranging a bit:

```suggestion
val view = LayoutInflater.from(context).inflate(R.layout.dialog_submit_survey_responses_success, null)
return Dialog(context, R.style.CenterFadeInOutDialogStyle).apply {
    window?.setBackgroundDrawable(ColorDrawable(android.graphics.Color.TRANSPARENT))
    setContentView(view)
}
```

Reply:
thank you for your suggestion, will update 🙏